src/client/editionsCommunication.ts

```diff
@@ -0,0 +1,93 @@
+// https://github.com/react-native-webview/react-native-webview/blob/master/src/WebViewTypes.ts#L963
+declare global {
+	interface Window {
+		ReactNativeWebView?: {
+			postMessage: (data: string) => void;
+		};
+
+		pingEditionsRendering?: (message: Message) => void;
+		pingEditionsRenderingJsString?: (message: Message) => string;
+	}
+}
+
+type ShareMessage = {
+	kind: 'share';
+};
+
+export type PlatformMessage = {
+	kind: 'platform';
+	value: Platform;
+};
+
+type PlatformQueryMessage = {
+	kind: 'platform-query';
+};
+
+type Message = ShareMessage | PlatformMessage | PlatformQueryMessage;
+
+export enum Platform {
+	IOS = 'IOS',
+	Android = 'Android',
+}
+
+export const isPlatformMessageEvent = (
+	customEvent: CustomEventInit<Message>,
+): customEvent is CustomEvent<PlatformMessage> => {
+	if (
+		customEvent.detail?.kind === 'platform' &&
+		customEvent.detail.value in Platform
+	) {
+		return true;
+	}
+	return false;
+};
+
+const prettyLog = (logMessage: string, data: string | Message): void => {
+	const parsedData: unknown =
+		typeof data === 'string' ? JSON.parse(data) : data;
+	console.log(
+		`%c${logMessage}
+${JSON.stringify(parsedData, null, 2)}`,
+		'color: #45969b;',
+	);
+};
+
+export const pingEditionsNative = (message: Message): void => {
+	const serializedMessage = JSON.stringify(message);
+
+	if (process.env.NODE_ENV === 'development') {
```

Comment:
Is `process` defined on the client?

Reply:
I always believed this is all stripped out in the production bundle thanks to the likes of webpack and friends. Now you said it, I will double check that this doesn't ship in production bundle 👍

Reply:
production bundle strips the whole things out 👍